### PR TITLE
3.1 - Add Mailer docs.

### PR DESCRIPTION
**Pull request by bcrowe**
Refs https://github.com/cakephp/cakephp/pull/6324

### CODE REVIEW

**Comment by jadb** [comment]
Lovely! I would also add that the `Mailer` implements the `EventListenerInterface`, explaining the advantages of that and maybe an example for that too.

Nice work!! Thanks!

File: en/core-libraries/email.rst [hunk 1]
```diff
@@ -429,6 +429,59 @@ message id (since there is no host name in a CLI environment)::
...
+
+    use App\Mailer\UserMailer;
+
+    // ...
+
+    $users = TableRegistry::get('Users');
+    $user = $users->get($id);
+
+    $mailer = new UserMailer;
+    $mailer->send('welcome', [$user]);
```
Comment (josegonzalez) [inline en/core-libraries/email.rst]: It get passed in via `call_user_func_array`?

Comment (bcrowe) [inline en/core-libraries/email.rst]: Yurp. https://github.com/cakephp/cakephp/pull/6324/files#diff-0d502f6f5b61a7cd1890732527bd7069R165

File: en/appendices/3-1-migration-guide.rst [hunk 2]
```diff
@@ -50,3 +50,15 @@ FlashHelper
 - ``FlashHelper`` can render multiple messages if multiple messages where
   set with the ``FlashComponent``. Each message will be rendered in its own
   element. Messages will be rendered in the order they were set.
+
+Email
+=====
+
+``Email`` and ``Transport`` classes have been moved under the ``Cake\Mailer``
+namespace.
```
Comment (markstory) [inline en/appendices/3-1-migration-guide.rst]: Should we mention that the old classnames are still available?

File: en/core-libraries/email.rst [hunk 3]
```diff
@@ -429,6 +429,59 @@ message id (since there is no host name in a CLI environment)::
...
+        {
+            $this->subject = 'Reset password';
+            $this->to = $user->email;
+            $this->set(['token' => $user->token]);
+        }
+    }
+
+In our example we have created two methods, one for sending a welcome email, and
+another for sending a password reset email. Each of these methods expect a user
+`Entity` and utilizes its properties for configuring each email.
```
Comment (markstory) [inline en/core-libraries/email.rst]: Need double backticks or a `:ref:` / `:doc:` link.

File: en/core-libraries/email.rst (hunk 1 above)
Comment (markstory) [inline en/core-libraries/email.rst]: Should you use the MailerAwareTrait to have a nicer example?

Comment (bcrowe) [inline en/core-libraries/email.rst]: The MailerAwareTrait doesn't seem to exist (yet). I saw people giving the thumbs up to merge the PR - so I thought it wasn't going to be made.

File: en/appendices/3-1-migration-guide.rst [hunk 4]
```diff
@@ -50,3 +50,16 @@ FlashHelper
 - ``FlashHelper`` can render multiple messages if multiple messages where
   set with the ``FlashComponent``. Each message will be rendered in its own
   element. Messages will be rendered in the order they were set.
+
+Email
+=====
+
+``Email`` and ``Transport`` classes have been moved under the ``Cake\Mailer``
+namespace. Their former namespaces are still usable as class aliases for them
+have been added.
```
Comment (ravage84) [inline en/appendices/3-1-migration-guide.rst]: > Their former namespaces are still usable as class aliases for them have been added.

I think that is one started sentence and another one ending :smirk_cat:

Comment (bcrowe) [inline en/appendices/3-1-migration-guide.rst]: Yeah, that sentence is fucked. :smirk:

**Comment by bcrowe** [comment]
Updated with `MailerAwareTrait` and event example.

File: en/core-libraries/email.rst [hunk 5]
```diff
@@ -429,6 +429,78 @@ message id (since there is no host name in a CLI environment)::
...
+
+    $users = TableRegistry::get('Users');
+    $user = $users->get($id);
+
+    $this->getMailer('User')->send('welcome', [$user]);
+
+If we wanted to completely separate sending a user their welcome email from our
+application's code, we can have our ``UserMailer`` subscribe to the
+``Model.afterSave`` event. By subcsribing to an event, we can keep our
+application's user-related classes completely free of email-related logic and
```
Comment (markstory) [inline en/core-libraries/email.rst]: Typo in subscribing.

**Comment by bcrowe** [comment]
Subscribing typo fixed.

**Comment by lorenzo** [comment]
ugh, cannot be automatically merged now :(

**Comment by cake17** [comment]
This can be closed in favor of #2874

**Pull request merged**